plugin/src/main/java/com/yelp/codegen/KotlinGenerator.kt

```diff
@@ -451,4 +455,30 @@ class KotlinGenerator : SharedCodegen() {
     override fun removeNonNameElementToCamelCase(name: String?): String {
         return super.removeNonNameElementToCamelCase(name, "[-_:;#\\[\\]]")
     }
+
+    /**
+     *  Function to check if there are Headers that should be applied at the Top level on Retrofit
+     *  with the @Headers annotation. This method will populate the `hasOperationHeaders` and `operationHeaders`
+     *  vendor extensions to support the mustache template.
+     */
+    internal fun processTopLevelHeaders(operation: CodegenOperation) {
+        val topLevelHeaders = mutableListOf<Pair<String, String>>()
+
+        // Send the X-Operation-Id header only if a custom operation ID was set.
+        val operationId = operation.vendorExtensions[X_OPERATION_ID] as String?
+        if (!operationId.isNullOrBlank()) {
+            topLevelHeaders.add(HEADER_X_OPERATION_ID to operationId)
+        }
+
+        // Send the Content-Type header for the first `consume` mediaType specified.
+        val firstContentType = operation.consumes?.firstOrNull()
```

Comment:
`consumes` might be defined on two different places:
 * `#/consumes` [doc](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#swagger-object): default consume for all the operations
 * `#/paths/<http_path>/<http_verb>/consumes` [doc](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#operation-object): consume override for the specific operation

does `operation.consumes` honours the default value in case the `consumes` defined in the operation is missing? If not, then we should be dealing with such case as well

Reply:
> does operation.consumes honours the default value in case the consumes defined in the operation is missing? If not, then we should be dealing with such case as well

I don't think that the top level `consumes` should be translated to a `@Header` annotation. Ideally a top level `consumes` information should be treated by an OkHttp interceptor at the client level.

Reply:
I manually verified the assumption and yes `CodegenOperation` does honours the top level `consumes` so this safe is good to go